examples/jsm/nodes/functions/ToonLightingModel.js

```diff
@@ -0,0 +1,51 @@
+import LightingModel from '../core/LightingModel.js';
+import BRDF_Lambert from './BSDF/BRDF_Lambert.js';
+import { diffuseColor } from '../core/PropertyNode.js';
+import { normalGeometry } from '../accessors/NormalNode.js';
+import { tslFn, float, vec2, vec3 } from '../shadernode/ShaderNode.js';
+import { mix, smoothstep } from '../math/MathNode.js';
+import { materialReference } from '../accessors/MaterialReferenceNode.js';
+
+const getGradientIrradiance = tslFn( ( { normal, lightDirection, builder } ) => {
+
+	// dotNL will be from -1.0 to 1.0
+	const dotNL = normal.dot( lightDirection );
+	const coord = vec2( dotNL.mul( 0.5 ).add(  0.5 ), 0.0 );
+
+	if ( builder.material.gradientMap ) {
+
+		const mr = materialReference( 'gradientMap', 'texture' );
```

Comment:
I think we can remove the hack for now using `context`: 
```js
const mr = materialReference( 'gradientMap', 'texture' ).context( { getUV: () => coord } );`
```

Reply:
much nicer - thanks